REGRESSION(298784@main): SHOULD NEVER BE REACHED in WebKit::DrawingAreaProxy::hideContentUntilAnyUpdate()

fujii opened the pull request:
#### 827dd33ab6e9c11b3512833584ead4b5805e92a8
<pre>
REGRESSION(298784@main): SHOULD NEVER BE REACHED in WebKit::DrawingAreaProxy::hideContentUntilAnyUpdate()
<a href="https://bugs.webkit.org/show_bug.cgi?id=297509">https://bugs.webkit.org/show_bug.cgi?id=297509</a>

Unreviewed crash fix for GTK port.
After &lt;<a href="https://commits.webkit.org/298784@main">https://commits.webkit.org/298784@main</a>&gt; changed to call
hideContentUntilAnyUpdate() unconditionally, GTK EWS layout tests and
GTK Debug layout tests are crashing due to an assertion failure in the
function.

* Source/WebKit/UIProcess/DrawingAreaProxy.h:
(WebKit::DrawingAreaProxy::hideContentUntilAnyUpdate):
Removed ASSERT_NOT_REACHED.

Canonical link: <a href="https://commits.webkit.org/298814@main">https://commits.webkit.org/298814@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3708d3ec97d23cb5cc49ad152dbbc8c89a6c6585

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/116764 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/36428 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/26997 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/122839 "Built successfully") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/67347 "Failed to checkout and rebase branch from PR 49504") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/37126 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/45029 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/88668 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/67347 "Failed to checkout and rebase branch from PR 49504") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/119713 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/29599 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/104741 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/69139 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/28661 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/66505 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/98978 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/23002 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/125974 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/43663 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/32786 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/97335 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/44027 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/100943 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/97133 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/42459 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/20394 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/39642 "Failed to checkout and rebase branch from PR 49504") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18642 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/43549 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/49144 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/43016 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/46355 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/44721 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/023ed1d56d6231e1c327ff39c54dbeedbd972a30. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49504

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3708d3ec97d23cb5cc49ad152dbbc8c89a6c6585. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49504

fujii committed:
REGRESSION(298784@main): SHOULD NEVER BE REACHED in WebKit::DrawingAreaProxy::hideContentUntilAnyUpdate()
https://bugs.webkit.org/show_bug.cgi?id=297509

Unreviewed crash fix for GTK port.
After <https://commits.webkit.org/298784@main> changed to call
hideContentUntilAnyUpdate() unconditionally, GTK EWS layout tests and
GTK Debug layout tests are crashing due to an assertion failure in the
function.

* Source/WebKit/UIProcess/DrawingAreaProxy.h:
(WebKit::DrawingAreaProxy::hideContentUntilAnyUpdate):
Removed ASSERT_NOT_REACHED.

Canonical link: https://commits.webkit.org/298814@main
